index.js

```diff
@@ -0,0 +1,281 @@
+"use strict";
+
+const Couchbase = require("couchbase");
+const { ViewQuery, N1qlQuery } = Couchbase;
+const debug = require("debug")("cb-utils:index");
+
+const noop = function () {};
+const delay = (ms) => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve();
+    }, ms);
+  });
+};
+
+function connectCluster(host, username, password) {
+  const cluster = new Couchbase.Cluster(host);
+  cluster.authenticate(username, password);
+  return cluster;
+}
+
+/**
+ * open bucket
+ * @param {Couchbase.Cluster} cluster
+ * @param {string} bucket
+ * @returns {Promise<Couchbase.Bucket>}
+ */
+function openBucket(cluster, bucket) {
+  return new Promise((resolve, reject) => {
+    const result = cluster.openBucket(bucket, (err) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(result);
+      }
+    });
+  });
+}
+
+function getDoc(bucket, key) {
+  return new Promise((resolve, reject) => {
+    bucket.get(key, (err, res) => {
```

Comment:
should skip `The key does not exist on the server` error and return null here?

Reply:
@superbogy why?